b2sdk/filter.py

```diff
@@ -0,0 +1,53 @@
+######################################################################
+#
+# File: b2sdk/filter.py
+#
+# Copyright 2024 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+from __future__ import annotations
+
+import fnmatch
+from enum import Enum
+from typing import NamedTuple, Sequence
+
+
+class FilterType(Enum):
+    INCLUDE = "include"
+    EXCLUDE = "exclude"
+
+
+class Filter(NamedTuple):
+    type: FilterType
+    pattern: str
+
+
+def include(pattern: str) -> Filter:
+    return Filter(type=FilterType.INCLUDE, pattern=pattern)
+
+
+def exclude(pattern: str) -> Filter:
+    return Filter(type=FilterType.EXCLUDE, pattern=pattern)
+
+
+class FilterMatcher:
```

Comment:
I believe this would benefit from a docstring. Especially since some of the behaviors are not obvious. For example, if an "include" filter and an "exclude" filter both match the same file, the filter that's listed later will take precedence. Or that no "exclude" filters results in excluding everything (except included object names).